src/server/ws-routes/controllers/Room.js

```diff
@@ -29,9 +30,9 @@ class Room {
 		const {roomId} = data
 		const username = socket.request.user.username
 		const timestamp = new Date().getTime()
-
-		socket.emit(EVENT_TYPES.JOIN, {username, roomId, timestamp})
+		socket.to(roomId).emit(EVENT_TYPES.JOIN, {username, roomId, timestamp})
 		socket.join(roomId)
```

Comment:
Swap `socket.to(roomId).emit(EVENT_TYPES.JOIN, {username, roomId, timestamp})` and `socket.join(roomId)`